src/main/java/com/blackducksoftware/integration/hub/polling/ScanStatusService.java

```diff
@@ -0,0 +1,169 @@
+		codeLocationRestService = new CodeLocationRestService(restConnection, gson, jsonParser);
+		scanSummaryRestService = new ScanSummaryRestService(restConnection, gson, jsonParser);
+		policyStatusRestService = new PolicyStatusRestService(restConnection, gson, jsonParser);
+	}
+
+	public PolicyStatusItem waitForCompleteScanStatus(final String groupId, final String artifactId,
+			final String version, final int timeoutInSeconds) throws IllegalArgumentException, URISyntaxException,
+			BDRestException, EncryptionException, IOException, UnexpectedHubResponseException, InterruptedException {
+		populateApplicableScansAndProjectVersions(groupId, artifactId, version);
+
+		final List<ScanSummaryItem> scanSummaries = new ArrayList<>();
+		for (final String scanSummaryLink : scanSummariesLinks) {
+			scanSummaries.addAll(scanSummaryRestService.getAllScanSummaryItems(scanSummaryLink));
+		}
+
+		final List<ScanSummaryItem> pendingScanSummaries = new ArrayList<>();
+		for (final ScanSummaryItem scanSummaryItem : scanSummaries) {
+			if (scanSummaryItem.getStatus().isPending()) {
+				pendingScanSummaries.add(scanSummaryItem);
+			}
+		}
+
+		waitUntilPendingScansAreComplete(pendingScanSummaries, timeoutInSeconds, System.currentTimeMillis());
+
+		final PolicyStatusItem policyStatusItem = policyStatusRestService.getItem(policyStatusLink);
+		return policyStatusItem;
+	}
+
+	private void populateApplicableScansAndProjectVersions(final String groupId, final String artifactId,
+			final String version)
+			throws IOException, BDRestException, URISyntaxException, UnexpectedHubResponseException {
+		final Map<String, Set<String>> mappedProjectVersionToScanSummariesLinks = new HashMap<>();
+		final List<CodeLocationItem> codeLocationItems = codeLocationRestService.getImportedCodeLocations(groupId,
+				artifactId, version);
+		for (final CodeLocationItem codeLocationItem : codeLocationItems) {
+			final String scanSummariesLink = codeLocationItem.getLink("scans");
+			final String mappedProjectVersionLink = codeLocationItem.getMappedProjectVersion();
+			if (!mappedProjectVersionToScanSummariesLinks.containsKey(mappedProjectVersionLink)) {
+				mappedProjectVersionToScanSummariesLinks.put(mappedProjectVersionLink, new HashSet<String>());
+			}
+
+			mappedProjectVersionToScanSummariesLinks.get(mappedProjectVersionLink).add(scanSummariesLink);
+		}
+
+		mappedProjectVersionLink = determineHubProjectAndVersionLink(mappedProjectVersionToScanSummariesLinks.keySet(),
+				artifactId, version);
+		if (StringUtils.isBlank(mappedProjectVersionLink)) {
+			throw new UnexpectedHubResponseException(
+					String.format("Couldn't find a mapped project/version for: %s %s.", artifactId, version));
+		}
+
+		scanSummariesLinks = mappedProjectVersionToScanSummariesLinks.get(mappedProjectVersionLink);
+	}
+
+	private void waitUntilPendingScansAreComplete(final List<ScanSummaryItem> pendingScanSummaries,
```

Comment:
Rather than recursive calls would using a FutureTask with the timeout set be possible.  You'd have to implement a Callable or runnable class to call waitUntilPendingScansAreComplete.  In waitUntilPendingScansAreComplete them implementation might be along the lines of:
`while(!pendingScanSummaries.isEmpty()) {

    Iterator<ScanSummaryItem> pendingItemIterator = pendingScanSummaries.iterator();  
    while(pendingItemIterator.hasNext()) {
        ScanSummaryItem item = pendingItemIterator.next();
        final String scanSummaryLink = item.getMeta().getHref();
	final ScanSummaryItem currentItem = scanSummaryRestService.getItem(scanSummaryLink);
        if (!currentItem.getStatus().isPending()) {
            pendingItemIterator.remove();
        }
    }
    Thread.sleep(FIVE_SECONDS);
}`